schema/equals.go

```diff
@@ -71,13 +75,16 @@ func (a Atom) Equals(b Atom) bool {
 	case a.List != nil:
 		return a.List.Equals(*b.List)
 	case a.Map != nil:
-		return a.Map.Equals(*b.Map)
+		return a.Map.Equals(b.Map)
```

Comment:
If we change Equals to take a pointer, then we should also change List for consistency (2 lines above)

Reply:
yeah, i agree. was just typing a comment to ask if you agree and i should add it :D

Reply:
But this means there are more changes to the Schema.

Reply:
How bad is it? So far it looks fine. I'd rather have a few more changes with consistency

Reply:
Depends, only List or the other ones as well (StructField, UnionField, Atom)

Reply:
Yeah, I can see that there are quite a few in `equals.go` ... I'd go for all of them :-/

Reply:
I'm doing that right now. This means changing a lot of types. So far no problems, but i'm not sure what errors this will cause. Let's see.
At least it gives me a reason to go through more of the smd codebase ;-)

Reply:
> At least it gives me a reason to go through more of the smd codebase ;-)

Exactly why I'm letting you do it ;-)